mitosheet/src/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -460,31 +477,57 @@ const GraphSidebar = (props: {
                             key={['yAxis'].concat(graphParams.yAxisColumnIDs).join('')}
                             columnIDsMap={props.columnIDsMapArray[selectedSheetIndex]}
                             columnDtypesMap={props.columnDtypesMap}
-                            
+
                             graphType={graphParams.graphType}
                             graphAxis={GraphAxisType.Y_AXIS}
                             selectedColumnIDs={graphParams.yAxisColumnIDs}
                             otherAxisSelectedColumnIDs={graphParams.xAxisColumnIDs}
-                            
+
                             updateAxisData={updateAxisData}
                             mitoAPI={props.mitoAPI}
                         />
+                        {props.sheetDataArray[selectedSheetIndex].numRows > 1000 &&
+                            <Row justify='space-between' align='center'>
+                                <Col>
+                                    <p className='text-header-3' title='Turning on Filter to Safe Size only graphs the first 1000 rows of your dataframe, ensuring that your browser won’t crash. Turning it off, graphs the entire dataframe and might crash your browser.'>
+                                        Filter to safe size
+                                    </p>
+                                </Col>
+                                <Col>
+                                    <Toggle
+                                        value={graphParams.safetyFilter}
+                                        onChange={toggleSafetyFilter}
+                                    />
+                                </Col>
+                            </Row>
+                        }
+                        {props.sheetDataArray[selectedSheetIndex].numRows <= 1000 &&
+                            <Row justify='space-between' align='center' title='Because you’re graphing less than 1000 rows of data, you’re browser tab will likely not crash.'>
+                                <Col>
+                                    <p className='text-header-3'>
+                                        Your data is safe to graph
+                                    </p>
+                                </Col>
+                                <SafeToGraphIcon />
+                            </Row>
+                        }
```

Comment:
Do you think it's better to display this or just say nothing? it almost feels like saying something in this case is... not ideal? On the other hand, users might be confused if the toggle is not there, so I see where you're coming from.

Perhaps the most clear thing to do here would be to disable the button if the data is to small, and default it to true? thoughts?